compiler/qsc_eval/src/intrinsic/utils.rs

```diff
@@ -39,12 +39,14 @@ pub fn split_state(
     }
 
     let dump_norm = 1.0 / dump_norm.sqrt();
-    Ok(dump_state
+    let mut dump_state = dump_state
         .into_iter()
         .filter_map(|(label, val)| {
             normalize_and_reorder(val, dump_norm, qubits, &label, qubit_count)
         })
-        .collect())
+        .collect::<Vec<_>>();
+    dump_state.sort_by(|(a, _), (b, _)| a.cmp(b));
```

Comment:
So the 'reorder' part of 'normalize_and_reorder' doesn't handle the sorting then?

Reply:
It reorders the bits in the label, but doesn't reorder the entries themselves. As in, `010` might become `100`, but something separately needs to look at all the reordered labels and sort the tuples by them.